src/graph.js

```diff
@@ -44,15 +44,52 @@ class Graph {
 			strings.push(resultSet.next().getString(0));
 		}
 		return strings;
-	}
+    }
+    
+    paramToString(paramValue) {
+        if(paramValue == null)
+            return "null"
+        let paramType = typeof(paramValue);
+        if(paramType == "string") {
+            let strValue = "";
+            if (paramValue[0] != "\"")
+                strValue += "\"";
+            strValue += paramValue;
+            if(paramValue[paramValue.length-1] != "\"")
+                strValue += "\"";
+            return strValue;
+        }
+        if(Array.isArray(paramValue))
+            return "[" + paramValue.toString() + "]";
+        return paramValue;
+    }
+
+    /**
+     * Extracts parameters from dictionary into cypher parameters string.
+     * 
+     * @param params parameters dictionary.
+     * @return a cypher parameters string.
+     */
+    buildParamsHeader(params) {
+        let paramsHeader = "CYPHER ";
+        for (var key in params) {
+            let value = this.paramToString(params[key]);
+            paramsHeader += `${key}=${value} `;
```

Comment:
Perhaps it's more efficient to use array.push() and then join() like StringBuilder in Java

Reply:
done